toolchain/parse/node_kind.def

```diff
@@ -683,7 +683,14 @@ CARBON_PARSE_NODE_KIND_CHILD_COUNT(ImplIntroducer, 0, Impl)
 //   _external_: ImplicitParamList
 // ImplForall
 CARBON_PARSE_NODE_KIND_CHILD_COUNT(ImplForall, 1, Forall)
-CARBON_PARSE_NODE_KIND_CHILD_COUNT(ImplAs, 0, As)
+
+// `... as`:
+//   _external_: expression
+// TypeImplAs
+CARBON_PARSE_NODE_KIND_CHILD_COUNT(TypeImplAs, 1, As)
+// `as` without a type before it
+CARBON_PARSE_NODE_KIND_CHILD_COUNT(DefaultSelfImplAs, 0, As)
+
 CARBON_PARSE_NODE_KIND_BRACKET(ImplDefinitionStart, ImplIntroducer,
```

Comment:
Should these parse nodes be more clearly associated with the comment that talks about them? Particularly as more is in the middle, I also wonder if the usual "external" would be better than "see below"